### ISSUE TITLE
tf.keras.layers.Dropout has no 'is_training' argument

### ISSUE THREAD

**Issue by jcRisch** [tensorflow/tensorflow#19420]
Dropout layer is (usually) only used while training. Keras automatically enable dropout layers while training and disable them while predicting.

Thus, tf.keras.layers.Dropout has no 'is_training' argument (but it exists in tf.layers.dropout). That can mislead Keras and tf users when trying to mix Keras with TF layers. Indeed, predicting new data with dropout layers doesn't produce any error but reduces the accuracy ! Maybe a warning note in 'tf.keras.layers.Dropout' documentation can be useful. 

With tf.layers
```
dropout = tf.layers.dropout(inputs=dense, rate=0.2, training=is_training, name="dropout-1")
```

With tf.Keras.layers
```
dense = ...
if(is_training):
      dropout = tf.keras.layers.Dropout(0.2, noise_shape=None, seed=None)(dense)
  else :
      dropout = dense
```

**Comment by henriquem87** [comment]
Hi @jcRisch,

I also struggled with this a bit before and I found out that tf.keras.layers.Dropout supports a training argument in its call function. Therefore, you could write:

```python
dropout = tf.keras.layers.Dropout(0.2, noise_shape=None, seed=None)(dense, training=is_training)
```

That also applies to tf.keras.layers.BatchNormalization (and possibly other layers).

I hope that helps.

**Comment by bhack** [comment]
The problems seems that the related doc is missing here `tf.keras` and upstream check https://github.com/keras-team/keras/issues/9412#issuecomment-366487734 /cc @fchollet

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 33 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 48 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 99 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 114 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by facaiy** [comment]
So we'd better to update Dropout's document, right? This seems like a good beginner issue.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by reactivetype** [comment]
Hi, I would like to contribute to this issue. May I work on it?

**Comment by I-Hong** [comment]
Is anyone working on this? If no, I'd like to add description about it.

**Comment by ianpdavies** [comment]
This still seems relevant because, as far as I know, the training argument does not exist in the Sequential API.

**Comment by MarkDaoust** [comment]
Sequential accepts a`training` argument now.

**Comment by carsonDB** [comment]
[The doc](https://www.tensorflow.org/guide/keras/custom_layers_and_models#privileged_training_argument_in_the_call_method) about `training` argument dropout is outdated.
Now `training` argument is a boolean tensor, not a python boolean.
So can't use python `if-else` only, instead actually use `tf.keras.backend.in_train_phase`, or [`smart_cond`](https://github.com/tensorflow/tensorflow/blob/r2.1/tensorflow/python/keras/layers/core.py#L170) to choose one of branches at the running time.

**Comment by Ashwin-Ramesh2607** [comment]
@carsonDB Hi, could you elaborate more, I am a beginner, and I am trying to create a custom layer by subclassing the `tf.keras.layers.Layers class ` 
Could you give an example of using `tf.keras.backend.in_train_phase` in the call() function of the custom layers class. I currently treat It as a python boolean and this is what I have written, please suggest the necessary changes to make.
`def call(self, inputs, training=None):`
    ` if training:
         output = tf.keras.layers.Dropout(self.dropout_rate)(inputs, training=training)`

**Comment by carsonDB** [comment]
@Ashwin-Ramesh2607 Hi, if you use standard `Dropout` layer, no need to judge if `training`, it handles already. If write custom dropout like I mentioned, your code can like this:
```
from tensorflow.python.keras import backend as K

def call(self, inputs, training=None):
    if training is None: # actually I test it in my tensorflow-1.15, `training` is always `None`.
          training = K.learning_phase()

    def dropped_inputs():
      return tf.nn.dropout(
          inputs,
          rate=self.dropout_rate)

    output = K.in_train_phase(dropped_inputs, inputs, training=training)
```
Since I don't test it, you'd better test it before using it.
